pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,406 @@
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// update the verbosity based on the existing verbosity in the KubeVirt CR
+	for key, value := range currentLv {
+		verbosityVal[key] = value
+	}
+
+	lines := createOutputLines(verbosityVal)
+
+	return lines
+}
+
+func addPatch(patchData *[]patch.PatchOperation, currentLv map[string]uint) {
+	*patchData = append(*patchData, patch.PatchOperation{
+		Op:    patchOperation,
+		Path:  patchPath,
+		Value: currentLv,
+	})
+}
+
+func setVerbosity(currentLv map[string]uint, patchData *[]patch.PatchOperation) {
+	// update currentLv based on the user-specified verbosity for all components
+	if *virtComponents[allComponent] != NoFlag {
+		for componentName := range virtComponents {
+			if componentName == allComponent {
+				continue
+			}
+			JSONName := getJSONNameByComponentName(componentName)
+			currentLv[JSONName] = *virtComponents[allComponent]
+		}
+	}
+
+	// update currentLv based on the user-specified verbosity for each component
+	for componentName, verbosity := range virtComponents {
+		if componentName == allComponent || *verbosity == NoFlag {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		currentLv[JSONName] = *verbosity
+	}
+
+	// in case of just reset (no set operation after the reset), don't need to add another patch
+	if len(currentLv) != 0 {
+		addPatch(patchData, currentLv)
+	}
+}
+
+func createPatch(currentLv map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+
+	// reset only if verbosity exists, otherwise do nothing
+	if isReset && len(currentLv) != 0 {
+		// add an empty object
+		currentLv = map[string]uint{}
+		addPatch(&patchData, currentLv)
+	}
+
+	setVerbosity(currentLv, &patchData)
+
+	return json.Marshal(patchData)
+}
+
+func findOperation(cmd *cobra.Command) (operation, error) {
+	isShow, isSet := false, false
+
+	for componentName, verbosity := range virtComponents {
+		// check if the flag for the component is specified
+		// cannot use NoFlag to check, because user can accidentally specify the same number as NoFlag for the verbosity
+		if !cmd.Flags().Changed(componentName) {
+			continue
+		}
+
+		// if flag is specified, it means either set or show
+		// if the value = noArg, it means show
+		// if the value != noArg, it means set
+		isShow = isShow || *verbosity == noArg
+		isSet = isSet || *verbosity != noArg
+
+		// check whether the verbosity is in the range
+		// Note that noArg is acceptable but operation is show instead of set
+		if *verbosity != noArg && *verbosity > maxVerbosity {
+			return nop, fmt.Errorf("%s: log verbosity must be %d-%d", componentName, minVerbosity, maxVerbosity)
+		}
+	}
+
+	// do not distinguish between set and reset at this point, because set and reset can coexist
+	if isReset {
+		isSet = true
+	}
+
+	if isShow && isSet {
+		return nop, errors.New("only show or set is allowed")
+	}
+
+	if isShow {
+		return show, nil
+	} else if isSet {
+		return set, nil
+	}
```

Comment:
nit: can this be refactored in a unique switch case?

Reply:
Yes. Will change it to the following.
```
	if isReset {
		isSet = true
	}
	switch {
	case isShow && isSet:
		return nop, errors.New("only show or set is allowed")
	case isShow:
		return show, nil
	case isSet:
		return set, nil
	default:
		return nop, nil
	}
```
`if isReset` clause has a different role than the other part. So, `if isReset` clause will be left as it is.

Reply:
Done